code/iaas/lb-logic/src/main/java/io/cattle/iaas/lb/service/impl/LoadBalancerServiceImpl.java

```diff
@@ -94,43 +96,33 @@ public void removeListenerFromConfig(LoadBalancerConfig config, long listenerId)
     }
 
     @Override
-    public void addTargetToLoadBalancer(LoadBalancer lb, long instanceId) {
-        LoadBalancerTarget target = lbTargetDao.getLbInstanceTarget(lb.getId(), instanceId);
-        if (target == null) {
-            target = objectManager.create(LoadBalancerTarget.class, LOAD_BALANCER_TARGET.INSTANCE_ID, instanceId,
-                    LOAD_BALANCER_TARGET.LOAD_BALANCER_ID, lb.getId(), LOAD_BALANCER_TARGET.IP_ADDRESS, null,
-                    LOAD_BALANCER_TARGET.ACCOUNT_ID, lb.getAccountId());
-        }
-        objectProcessManager.scheduleProcessInstance(LoadBalancerConstants.PROCESS_LB_TARGET_MAP_CREATE,
-                target, null);
+    public void addTargetToLoadBalancer(LoadBalancer lb, LoadBalancerTargetInput targetInput) {
```

Comment:
Since you are refactoring some of this code now, is it possible to split out the methods called in the API tier from the backend tier?  So LoadBalancerServiceImpl and LoadBalancerApiServiceImpl or something similar.  Given all the troubles with DeferredUtils it would be good to cleanly separate the code.